ui/page/components/wallet_sync_info.go

```diff
@@ -541,6 +543,10 @@ func (wsi *WalletSyncInfo) ListenForNotifications() {
 		// headers to fetch cannot be less than the previously fetched.
 		// Page refresh only needed if there is new data to update the UI.
 		if progress.HeadersToFetchOrScan >= previousProgress.HeadersToFetchOrScan {
+			// Lock the map before updating it
+			wsi.syncProgressInfoMutex.Lock()
+			// Ensure the mutex is unlocked even if there is a panic
+			defer wsi.syncProgressInfoMutex.Unlock()
```

Comment:
Panic from where?

Reply:
i guess that comment is not necessary, would remove in one of my upcoming PRs